### ISSUE TITLE
Unable to deserialize Source object as [FromBody] parameter

### ISSUE THREAD

**Issue by goldenass** [stripe/stripe-dotnet#1361]
Updating from 19.10.0 passing a **[FromBody] Source param** in a webapi controller results in a Newtonsoft deserialization error, trying to deserialize the filed **created**. For what I can see the problem maybe the fact that in version 20.+ the part:

```c#
[JsonConverter(typeof(StripeDateTimeConverter))]
[JsonProperty("created")]
public DateTime Created { get; set; }
```

changed to:

```c#
[JsonProperty("created")]
public DateTime Created { get; set; }
```

Is it a bug, or there exists another preferred way t do the same thing (Sending a Source object as a [FromBody] json parameter)?

**Comment by ob-stripe** [comment]
Hi @goldenass. This change was intended, as the `DateTime` converter is now automatically applied internally. However, we didn't realize that some users such as yourself were manually deserializing resources.

We're considering reverting that change or providing some other way to manually deserialize resources. We'll update here once we have a fix. Sorry for the inconvenience!

**Comment by goldenass** [comment]
Ok good. Thank you very much!

**Comment by ob-stripe** [comment]
Quick update: unfortunately, properly solving this issue is more complex than I anticipated, due to the fact that we've replaced the "intermediary" classes that we used for polymorphic resources (e.g. [`BalanceTransactionSource`](https://github.com/stripe/stripe-dotnet/blob/v19.10.0/src/Stripe.net/Entities/BalanceTransactionSource.cs#L22)) with interfaces (e.g. [`IBalanceTransactionSource`](https://github.com/stripe/stripe-dotnet/blob/v20.4.0/src/Stripe.net/Entities/_interfaces/IBalanceTransactionSource.cs#L6)).

We cannot apply the converter directly to the interface, otherwise the converter is applied every time a class that implements the interface is deserialized (whereas we only want to use the converter when deserializing an attribute that is explicitly declared with the interface type, e.g. [`BalanceTransaction.Source`](https://github.com/stripe/stripe-dotnet/blob/v20.4.0/src/Stripe.net/Entities/BalanceTransactions/BalanceTransaction.cs#L47)).

We also cannot apply the converter directly to the property because of the `StringOrObject` indirection layer used to handle attribute expansion.

Properly solving this will likely require updating the logic for expandable attributes to call the correct converter, but I'm unsure yet how that would work exactly (maybe manually looking up the `JsonConverter` attribute somehow?).

In the meantime, I'm thinking of making our (currently internal) converters public, so that users can at least do something like `JsonConvert.Deserialize<SomeClass>(json, Stripe.JsonSettings)` to manually deserialize JSON strings into resource classes. That wouldn't work for the usecase described in this issue though (i.e. using [`[FromBody]`](https://docs.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/parameter-binding-in-aspnet-web-api#using-frombody)).
